### DISCUSSION TITLE
Custom Function JSON Response Handling

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2085] [Q&A]
I am tring to return response as from custom function as 

> return c.JSON(200, finalRes)
Interface Structure is 
struct {
	Status  bool   `json:"status"`
	Message string `json:"message"`
}

In flutter SDK returning error as "Expected a value of type 'String?', but got one of type '_JsonMap'" but in network I am receiving proper response as in below format
{
    "status": true,
    "message": ""
}

How to handle this?

I tried to change it as string and returned c.String(cast.ToString(finalRes)), In this case flutter sdk not throw any error but pb.send return value is null.

**Comment by ganigeorgiev** [comment]
Could you provide a more detailed code sample for both the custom Go route and how you are sending the request with the Dart SDK?

**Reply by iamdineshkumar** [reply]
In go route
```

type CustStatusResponse struct {
	Status  bool   `json:"status"`
	Message string `json:"message"`
}
finalRes.Status = false
finalRes.Message = "Transaction Failed"
return c.JSON(200, finalRes)
```

In dart code
pb.send('api/setamount',method: "POST",body: {"id": id,"amt": 1000},headers: {"Content-Type":"application/json; charset=UTF-8"},);

**Reply by ganigeorgiev** [reply]
The Go code code looks ok, I don't think there is anything unusual. The Dart code on the other hand is incomplete and it is hard to say what triggers the "_Expected a value of type 'String?_" error?

My guess is that the error is not related to the response but to something in how the send `Future` result is handled in the Flutter app, but the above Dart code is incomplete and doesn't show how you are handling the response.

Here is a sample Dart `pb.send` call that should work:

```dart
pb.send(
  "/api/setamount",
  method: "POST",
  body: {"id": "123", "amt": 1000},
).then((data) {
  print(data as Map<String, dynamic>? ?? {});
});
```
